lib/game.ex

```diff
@@ -17,39 +17,54 @@ defmodule Game do
   defp convert_to_rows(board), do: Enum.chunk(board, 3)
 
   def get_winner(game) do
-    winner =
-    game.board
-      |> convert_to_diagonals
-      |> check_line
-      |> Enum.find(&(&1)) ||
-    game.board
-      |> convert_to_rows
-      |> check_line
-      |> Enum.find(&(&1)) ||
-    game.board
-      |> convert_to_columns
-      |> check_line
-      |> Enum.find(&(&1)) 
-
+    diagonals = 
+      game.board
+        |> convert_to_diagonals
+        |> check_line
+        |> Enum.find(&(&1))
+    rows = 
+      game.board
+        |> convert_to_rows
+        |> check_line
+        |> Enum.find(&(&1))
+    columns =
+      game.board
+        |> convert_to_columns
+        |> check_line
+        |> Enum.find(&(&1)) 
+    winner = diagonals || rows || columns
     %{game | winner: winner} 
   end
 
   def is_tie?(game), do: Board.available_spaces(game.board) |> length == 0
 
   def game_over?(game) do  
     cond do
-      get_winner(game).winner() ->
-        true
-      is_tie?(game) ->
-        true
-      true ->
-        false
+      get_winner(game).winner() -> true
+      is_tie?(game) -> true
+      true -> false
     end
   end
 
-  def change_turn(game, %Player{}), do: %{ game | current_player: %Computer{}}
-  def change_turn(game, %Computer{}), do: %{ game | current_player: %Player{}}
+  def change_turn(game, %Player{}), do: update(game, :current_player, %Computer{})
+  def change_turn(game, %Computer{}), do: update(game, :current_player, %Player{})
 
+  def update(game, key, value) do
+    cond do
+      :current_player == key ->
+        %{ game | current_player: value}
+      :player_1 == key ->
+        %{ game | player_1: value}
+      :player_2 == key ->
+        %{ game | player_2: value}
+      :current_player == key ->
+        %{ game | current_player: value}
+      :winner == key ->
+        %{ game | winner: value}
+      :board == key ->
+        %{ game | board: value}
```

Comment:
How would you feel about this function if it were pattern matching against arguments?  It seems like in all cases you're checking on `key`.  What are your thoughts on this form?

```elixir
def update(game, :current_player, current_player), do: %{ game | current_player: current_player }
def update(game, :player_1, player_1), do: %{ game | player_1: player_1 }
...etc...
```